tests/run-make/extern-fn-reachable/Makefile

```diff
@@ -1,26 +0,0 @@
-# ignore-cross-compile
-include ../tools.mk
-
-# ignore-windows-msvc
-
-NM=nm -D
-
-ifeq ($(UNAME),Darwin)
-NM=nm -gU
-endif
-
-ifdef IS_WINDOWS
-NM=nm -g
-endif
```

Comment:
Question: Hm,

- `-D` is "display the dynamic symbols rather than the normal symbols. This is only meaningful for dynamic objects, such as certain types of shared libraries"
- `-g` is "display only external symbols"
- `-U` is "display only defined symbols for each object file. By default both defined and undefined symbols are displayed"

and these flags are passed differently based on platforms. Are we sure `with_symbol_iter` based on `object` preserves these behavior?

Reply:
the test is configured to never run on those platforms anyways.

Reply:
> the test is configured to never run on those platforms anyways.

Hmm? That doesn't seem right, since the Makefile test is only `ignore-cross-compile` + `ignore-windows-msvc`, or am I misunderstanding something? This still leaves host linux, host apple and host windows mingw targets?

Reply:
It seems you're correct, I must've been thinking of a different rmake test.

Still, it seems strange that it would look for different kinds of symbols on different platforms, shouldn't the symbols have the same linkage regardless of platform?

Reply:
> Still, it seems strange that it would look for different kinds of symbols on different platforms, shouldn't the symbols have the same linkage regardless of platform?

Yeah, this is exactly why I noticed it. I'll try to do a little digging from the original PR to see why it had to use different `nm` flags, this seems very cursed.

Reply:
there's the `ObjectSymbol::is_*` family of function, but i'm somewhat confused what it would mean to have a non-dynamic global symbol in a dynamic object.